expanding on date histogram docs (#11149)
opensearch-project/documentation-website: _aggregations/bucket/date-histogram.md

```json
GET my-logs/_search
{
  "size": 0,
  "aggs": {
    "by_hour": {
      "date_histogram": {
        "field": "timestamp",
        "fixed_interval": "1h"
      }
    }
  }
}
```
{% include copy-curl.html %}

## Example: Use a time zone

By default, bucketing occurs in UTC. Set `time_zone` to align bucket boundaries to a specific time zone.

Retrieve daily buckets using `Europe/Dublin`:

```json
GET my-logs/_search
{
  "size": 0,
  "aggs": {
    "by_day_ie": {
      "date_histogram": {
        "field": "timestamp",
        "calendar_interval": "day",
        "time_zone": "Europe/Dublin"
      }
    }
  }
}
```
{% include copy-curl.html %}

## Example: Shift bucket start times using an `offset`

Use the `offset` parameter to move the bucket boundary forward or backward. For example, to define a "reporting day" that runs 06:00–06:00 instead of midnight–midnight:

```json
GET my-logs/_search
{
  "size": 0,
  "aggs": {
    "by_day_shifted": {
      "date_histogram": {
        "field": "timestamp",
        "calendar_interval": "day",
        "offset": "+6h"
      }
    }
  }
}
```
{% include copy-curl.html %}

## Example: Include empty buckets

> Set `min_doc_count` to `0` and provide a range in `extended_bounds` to return empty buckets across the whole time window.

Reviewer: ```suggestion
Set `min_doc_count` to `0` and provide a range in `extended_bounds` to return empty buckets across the entire time window.
```

Retrieve one-hour buckets for the last 24 hours, including hours with no data:

```json
GET my-logs/_search
{
  "size": 0,
  "aggs": {
    "last_24h": {
      "date_histogram": {
        "field": "timestamp",
        "fixed_interval": "1h",
        "min_doc_count": 0,
        "extended_bounds": {"min": "now-24h", "max": "now"}
      }
    }
  }
}
```
{% include copy-curl.html %}

## Example: Limit the range strictly

`hard_bounds` strictly limits the histogram to the specified minimum and maximum time range. No buckets are created outside these bounds, even if data falls beyond them.

> Retrieve 30-minute buckets for the period between `2025-09-01T00:00:00Z` and `2025-09-01T06:00:00Z`:

Reviewer: ```suggestion
Retrieve buckets with a fixed interval of 30 minutes for the period between `2025-09-01T00:00:00Z` and `2025-09-01T06:00:00Z`:
```

```json
GET my-logs/_search
{
  "size": 0,
  "aggs": {
    "strict_range": {
      "date_histogram": {
        "field": "timestamp",
        "fixed_interval": "30m",
        "hard_bounds": {"min": "2025-09-01T00:00:00Z", "max": "2025-09-01T06:00:00Z"}
      }
    }
  }
}
```
{% include copy-curl.html %}

## Example: Return a map of buckets using `keyed`

> Set `keyed: true` to return buckets as an object keyed by the formatted date string:.

Reviewer 2: ```suggestion
Set `keyed: true` to return buckets as an object keyed by the formatted date string:
```

```json
GET my-logs/_search
{
  "size": 0,
  "aggs": {
    "by_hour_script": {
      "date_histogram": {
        "script": {
          "lang": "painless",
          "source": "return doc['timestamp'].value.toInstant().toEpochMilli();"
        },
        "fixed_interval": "1h"
      }
    }
  }
}
```
{% include copy-curl.html %}

## Parameters

The `date_histogram` supports the following parameters.

| Parameter | Required | Type | Description |
|:--|:--|:--|:--|
| `field` | One of the following is required: `field` or `script` | String | The date/datetime field to bucket on. |
| `calendar_interval` | One of the following is required: `calendar_interval`, `fixed_interval` or legacy `interval` | String | The calendar-aware interval (for example, `"day"`, `"1M"`, `"year"`). Only singular calendar units are supported. |

> | `fixed_interval` | One of the following is required: `calendar_interval`, `fixed_interval` or legacy `interval` | String | The exact-duration interval, for example, `"5m"`, `"12h"`, `"30d"`. Not for calendar units like months or quarters. |

Reviewer: ```suggestion
| `fixed_interval` | One of the following is required: `calendar_interval`, `fixed_interval`, or legacy `interval`. | String | The fixed interval, for example, `"5m"`, `"12h"`, `"30d"`. Not for calendar units like months or quarters. |
```

| `time_zone` | Optional | String | The time zone used for bucketing and formatting. Accepts timezone, such as `"Europe/Dublin"` or UTC offsets, such as `"-07:00"`. |
| `format` | Optional | String | The output date format used for `key_as_string`, for example, `"yyyy-MM-dd"`. If omitted, mapping defaults apply. |
| `offset` | Optional | String | Shifts bucket boundaries by a positive or negative duration, for example, `"+6h"`, `"-30m"`. Calculated after `time_zone` is applied. |

> | `min_doc_count` | Optional | Integer | The minimum number of docs required to return a bucket. Default is `1`. Set to `0` to include empty buckets. |

Reviewer: ```suggestion
| `min_doc_count` | Optional | Integer | The minimum number of documents required in order to return a bucket. Default is `1`. Set to `0` to include empty buckets. |
```

| `extended_bounds` | Optional | Object | Extends the range of buckets beyond your data: `{"min": "<date>", "max": "<date>"}`. Often used with `min_doc_count: 0`. |
| `hard_bounds` | Optional | Object | Strictly limits buckets to a range: `{"min": "<date>", "max": "<date>"}`. Buckets outside the range are never created. |
| `missing` | Optional | Date string | Treat docs missing the field as if they had this date value. |
| `keyed` | Optional | Boolean | When `true`, returns buckets as an object keyed by the formatted date string. |
| `order` | Optional | Object | Sort buckets by `_key` or `_count`, ascending or descending. |

> | `script` | One of the following is required: `field` or `script` | Object | Optional script to compute the value to bucket on. Use with care; scripts add overhead. |

Reviewer 3: Optional script to compute the value to bucket on. Since the scripts are operated to modify each value, they add overhead and should be used cautiously.
